clients/da-vinci-client/src/main/java/com/linkedin/davinci/notifier/PushStatusNotifier.java

```diff
@@ -116,16 +120,28 @@ public void dataRecoveryCompleted(String kafkaTopic, int partitionId, long offse
 
   @Override
   public void startOfIncrementalPushReceived(String topic, int partitionId, long offset, String message) {
-    offLinePushAccessor
-        .updateReplicaStatus(topic, partitionId, instanceId, START_OF_INCREMENTAL_PUSH_RECEIVED, offset, message);
-    updateIncrementalPushStatusToPushStatusStore(topic, message, partitionId, START_OF_INCREMENTAL_PUSH_RECEIVED);
+    updateIncrementalPushStatus(topic, partitionId, offset, message, START_OF_INCREMENTAL_PUSH_RECEIVED);
   }
 
   @Override
   public void endOfIncrementalPushReceived(String topic, int partitionId, long offset, String message) {
-    offLinePushAccessor
-        .updateReplicaStatus(topic, partitionId, instanceId, END_OF_INCREMENTAL_PUSH_RECEIVED, offset, message);
-    updateIncrementalPushStatusToPushStatusStore(topic, message, partitionId, END_OF_INCREMENTAL_PUSH_RECEIVED);
+    updateIncrementalPushStatus(topic, partitionId, offset, message, END_OF_INCREMENTAL_PUSH_RECEIVED);
+  }
+
+  private void updateIncrementalPushStatus(
+      String topic,
+      int partitionId,
+      long offset,
+      String message,
+      ExecutionStatus status) {
+    if (incrementalPushStatusWriteMode == IncrementalPushStatusWriteMode.ZOOKEEPER_ONLY
+        || incrementalPushStatusWriteMode == IncrementalPushStatusWriteMode.DUAL) {
+      offLinePushAccessor.updateReplicaStatus(topic, partitionId, instanceId, status, offset, message);
+    }
+    if (incrementalPushStatusWriteMode == IncrementalPushStatusWriteMode.PUSH_STATUS_SYSTEM_STORE_ONLY
+        || incrementalPushStatusWriteMode == IncrementalPushStatusWriteMode.DUAL) {
+      updateIncrementalPushStatusToPushStatusStore(topic, message, partitionId, status);
+    }
```

Comment:
Can you update the below batch update part?

Reply:
Done